src/libs/actions/PersonalDetails/index.js

```diff
@@ -237,20 +237,21 @@ function setPersonalDetails(details) {
 }
 
 /**
- * Fetches the preferred currency for the current user
- *
+ * Fetches the Currency preferences based on location
+ * @param {bool} onLocation
  * @param {Object} details
  */
-function fetchCurrencyPreferences() {
+function fetchCurrencyPreferences(onLocation) {
     let coords = {};
     let currency = '';
-
-    Geolocation.getCurrentPosition((position) => {
-        coords = {
-            longitude: position.coords.longitude,
-            latitude: position.coords.latitude,
-        };
-    });
+    if (onLocation) {
+        Geolocation.getCurrentPosition((position) => {
+            coords = {
+                longitude: position.coords.longitude,
+                latitude: position.coords.latitude,
+            };
+        });
+    }
```

Comment:
So this is to prevent web from requesting location on start?

Reply:
Yeah. It is because we need preferences fetched for the user in the startup, otherwise the IOUModal will get undefined selectedCurrency. So, I set up this to work this way: 

Normally, we wouldn't ask user for the permission. 

When user goes to the Currency Page, I refetch preferences in ComponentDidMount so we can get preferences using the location.